client.go

```diff
@@ -553,6 +553,12 @@ func (c *Client) Subscribe(params *SubscriptionParameters) (*Subscription, error
 		c,
 	}
 	c.subMux.Lock()
+	if _, ok := c.subscriptions[sub.SubscriptionID]; ok {
```

Comment:
`SubscriptionID` is an `IntegerId` (Part 4, 7.14) which cannot be zero. So we should also check for `0` and I think you can simplify the check to 

```go
if c.subscriptions[sub.SubscriptionID] != nil {
	c.subMux.Unlock()
	return sub, ua.StatusBadSubscriptionIDInvalid
}
```